.github/workflows/go-mod-opt.yml

```diff
@@ -0,0 +1,80 @@
+name: go mod cache
+env:
+  PRIVATE_KEY: ${{ secrets.AWS_PRIVATE_KEY  }}
+  TERRAFORM_AWS_ASSUME_ROLE: ${{ secrets.TERRAFORM_AWS_ASSUME_ROLE }}
+  TERRAFORM_AWS_ASSUME_ROLE_DURATION: 14400 # 4 hours
+  S3_INTEGRATION_BUCKET: ${{ secrets.S3_INTEGRATION_BUCKET }}
+  KEY_NAME: ${{ secrets.KEY_NAME }}
+  CF_IAM_ROLE: ${{ secrets.CF_IAM_ROLE }}
+  CF_KEY_NAME: ${{ secrets.CF_KEY_NAME }}
+  CACHE_EXIST: "false"
+  CWA_GITHUB_REPO_NAME: "aws/amazon-cloudwatch-agent"
+  #  S3_INTEGRATION_BUCKET: "terraform-integration-test"
+  #  CACHE_DIR: "uniform-build-env-test/cache_build_env"
+  CACHE_DIR: "s3://cloudwatch-agent-integration-bucket/cloudwatch-agent-integration-bucket/logs/"
+on: push
+jobs:
+  cache:
+    name: Go mod cache
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ secrets.TERRAFORM_AWS_ASSUME_ROLE }}
+          aws-region: us-west-2
+      - uses: actions/checkout@v3
+        with:
+          repository: ${{env.CWA_GITHUB_REPO_NAME}}
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v2
+        with:
+          go-version: ~1.20.5
+      - name: set up AWS CLI
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
+      - name: go mod optimization
+        run: |
+          if aws s3 ls "s3://cloudwatch-agent-integration-bucket/vendorTest/vendor.zip"; then
+             echo "time for copy file to vendor from s3 bucket"
+             time aws s3 cp "s3://cloudwatch-agent-integration-bucket/vendorTest/vendor.zip" ./vendor.zip
+             echo "time for unzipping the vendor file"
+             time unzip vendor.zip
+             rm -r vendor.zip
+             set +e
+             echo "building the cloudwatch agent"
+             time make build GOMODCACHE=true
+             build_exit_status=$?
+             echo  "build status code:$build_exit_status"
+             if [ $build_exit_status -ne 0 ]; then   #To handle the updated mod file
```

Comment:
i dont understand how failing to build the agent represent new modules inside the go.mod. Is this because you are forcing vendor in Makefile so if vendor file don't exist the build fails? Can you clarify this?